tests/test_migrate_coverage.py

```
@@ -0,0 +1,1014 @@
+
+        # Mock only the backup call, not the actual migration copy
+        original_copy2 = shutil.copy2
+        call_count = [0]
+
+        def selective_copy2(src, dst, *args, **kwargs):
+            call_count[0] += 1
+            # First call should be backup (dest exists)
+            if call_count[0] == 1 and "meshtastic.sqlite" in dst and call_count[0] == 1:
+                # Check if we're backing up existing (not copying from legacy)
+                if "new_home" in str(dst):
+                    raise OSError("Mock backup error")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)